ConsoleApp/ConsoleApp/Program.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+
+namespace ConsoleApp
+{
+    class Program
+    {
+        static void Main()
+        {
+            //F and M methods
+            Console.WriteLine("F and M methods");
+            Console.WriteLine(F(5));
+            Console.WriteLine(M(4));
+            //Pow method
+            Console.WriteLine("Pow method");
+            Console.WriteLine(Pow(2, 5));
+
+            //AllNumbers method
+            Console.WriteLine("AllNumbers method");
+            AllNumbers(5);
+        }
+
+        //F and M methods
+        static int F(int n)
+        {
+            if (n == 0) return 1;
+            return n - M(F(n - 1));
+        }
+        static int M(int n)
+        {
+            if (n == 0) return 0;
+            return n - F(M(n - 1));
+        }
+
+        //Pow method
+        static int Pow(int x, int y)
+        {
+            if (y == 0) return 1;
+            if (y == 1) return x;
```

Comment:
you can even skip this condition because you have branch above, but anyway it works